### ISSUE TITLE
Move serde 1.0 support in-crate

### ISSUE THREAD

**Issue by brson** [servo/rust-url#304]
Today serde 1.0 is supported in its own `url_serde` crate, so that url can support both old serde and new serde. @dtolnay has some plan to hack support for both in-crate, though presumably serde_url will need to continue on until a major bump.

**Comment by SimonSapin** [comment]
Removing support for serde 0.8 would be a breaking change, which I’d rather not make until there are other and strong enough reasons to do so. Therefore this depends on #320.

`serde_url` will still be on crates.io, available to projects that already use it.

**Comment by dtolnay** [comment]
#327 adds support in-crate for Serde 1.0. Should I make a separate PR removing `url_serde` from the repo?

**Comment by drusellers** [comment]
Looking forward to serde 1.0 support landing

**Comment by opilar** [comment]
Is this still relevant? I would like to help.

**Comment by SimonSapin** [comment]
I’d prefer to be able to use `derive`, which depends at least on https://github.com/serde-rs/serde/issues/953.

**Comment by silwol** [comment]
Any news on this?
In Debian, we have packaged `url`, but the `serde` feature and it's dependency to the pre-1.0 version of `serde` prevents it's migration from unstable to testing.
We have some options there:
- Patch out the `serde` feature from `url` (preventing any dependencies from packaging that use this feature).
- Wait until an updated `url` crate without pre-1.0 `serde` dependency is released, and get other crates which depend on it to upgrade (most tedious, but probably the most sustainable course of action).
- Package `serde 0.8` (which we don't intend to do).

**Comment by SimonSapin** [comment]
No real news for now. As noted above, this presumably requires dropping support for older serde version and so bumping the version number to 2.0.0. I’ve filed https://github.com/servo/rust-url/issues/463 about that, but as you can see there’s potentially a bunch of related work that no-one else is actively doing at the moment.

**Comment by SimonSapin** [comment]
@silwol Serde is an optional dependency. Does Debian require optional dependencies to be packaged as well even if nothing uses them?

**Comment by jimmycuadra** [comment]
Any progress on this?

**Comment by nox** [comment]
Done in master. Will be released in 2.0.
